src/java/org/apache/cassandra/service/reads/ReadCallback.java

```diff
@@ -56,29 +59,104 @@
 public class ReadCallback<E extends Endpoints<E>, P extends ReplicaPlan.ForRead<E>> implements RequestCallback<ReadResponse>
 {
     protected static final Logger logger = LoggerFactory.getLogger( ReadCallback.class );
-    private class WarningCounter
+
+    private class WarnAbortCounter
     {
-        // the highest number of tombstones reported by a node's warning
-        final AtomicInteger tombstoneWarnings = new AtomicInteger();
-        final AtomicInteger maxTombstoneWarningCount = new AtomicInteger();
-        // the highest number of tombstones reported by a node's rejection. This should be the same as
-        // our configured limit, but including to aid in diagnosing misconfigurations
-        final AtomicInteger tombstoneAborts = new AtomicInteger();
-        final AtomicInteger maxTombstoneAbortsCount = new AtomicInteger();
-
-        // TODO: take message as arg and return boolean for 'had warning' etc
-        void addTombstoneWarning(InetAddressAndPort from, int tombstones)
+        final AtomicInteger warnings = new AtomicInteger();
+        // the highest number reported by a node's warning
+        final AtomicLong maxWarningCount = new AtomicLong();
```

Comment:
nit: maybe `maxWarningValue`? (Anyone would figure it's not that, given the line above, but I first read this as "number of warnings".)